docs/sophia_stdlib.md

```diff
@@ -488,2003 +909,1586 @@ Oracle.check_query(o : oracle('a, 'b), q : oracle_query('a, 'b)) : bool
 
-Rounds a fraction to a nearest integer. If two integers are in the same distance it
-will choose the even one.
+##### subtract
 
+```
+Set.subtract(s1 : set('a), s2 : set('a)) : set('a)
+```
 
-##### add
-`Frac.add(a : frac, b : frac) : frac`
+Returns the elements of `s1` that are not members of `s2`
 
-Sum of the fractions.
+##### intersection
 
+```
+Set.intersection(s1 : set('a), s2 : set('a)) : set('a)
+```
 
-##### neg
-`Frac.neg(a : frac) : frac`
+Returns the intersection of the two sets `s1` and `s2`
 
-Negation of the fraction.
+##### intersection_list
 
+```
+Set.intersection_list(sets : list(set('a))) : set('a)
+```
 
-##### sub
-`Frac.sub(a : frac, b : frac) : frac`
+Returns the intersection of all the sets in the given list
 
-Subtraction of two fractions.
+##### union
 
+```
+Set.union(s1 : set('a), s2 : set('a)) : set('a)
+```
 
-##### inv
-`Frac.inv(a : frac) : frac`
+Returns the union of the two sets `s1` and `s2`
 
-Inverts a fraction. Throws error if `a` is zero.
+##### union_list
 
+```
+Set.union_list(sets : list(set('a))) : set('a)
+```
 
-##### mul
-`Frac.mul(a : frac, b : frac) : frac`
+Returns the union of all the sets in the given list
 
-Multiplication of two fractions.
 
+### String
 
-##### div
-`Frac.div(a : frac, b : frac) : frac`
+Operations on the `string` type. A `string` is a UTF-8 encoded byte array.
 
-Division of two fractions.
+#### length
+`length(s : string) : int`
 
+The length of a string.
 
-##### int_exp
-`Frac.int_exp(b : frac, e : int) : frac`
+Note: not equivalent to byte size of the string, rather `List.length(String.to_list(s))`
 
-Takes `b` to the power of `e`. The exponent can be a negative value.
+#### concat
+```
+concat(s1 : string, s2 : string) : string
+```
 
+Concatenates `s1` and `s2`.
 
-##### optimize
-`Frac.optimize(f : frac, loss : frac) : frac`
+#### concats
+```
+concats(ss : list(string)) : string
+```
 
-Shrink the internal size of a fraction as much as possible by approximating it to the
-point where the error would exceed the `loss` value.
+Concatenates a list of strings.
 
+#### to\_list
+```
+to_list(s : string) : list(char)
+```
 
-##### is_sane
-`Frac.is_sane(f : frac) : bool`
+Converts a `string` to a list of `char` - the code points are normalized, but
+composite characters are possibly converted to multiple `char`s. For example the
+string "😜i̇" is converted to `[128540,105,775]` - where the smiley is the first
+code point and the strangely dotted `i` becomes `[105, 775]`.
 
-For debugging. If it ever returns false in a code that doesn't call `frac` constructors or
-accept arbitrary `frac`s from the surface you should report it as a
-[bug](https://github.com/aeternity/aesophia/issues/new)
+#### from\_list
+```
+from_list(cs : list(char)) : string
+```
 
-If you expect getting calls with malformed `frac`s in your contract, you should use
-this function to verify the input.
+Converts a list of characters into a normalized UTF-8 string.
 
-## <a name='set-stdlib'>Set</a>
```

Comment:
I have added this `<a name='set-stdlib'>` because without it `#set` will direct you to `Bits.set` instead of `Set`

Reply:
yeah, should now work again with correct structure